test/new-e2e/examples/vm_fakeintake_test.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package examples
+
+import (
+	_ "embed"
+	"errors"
+	"time"
+
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	ec2vm "github.com/DataDog/test-infra-definitions/aws/scenarios/vm/ec2VM"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
+	commonos "github.com/DataDog/test-infra-definitions/common/os"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type vmFakeintakeSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &vmFakeintakeSuite{}, e2e.AgentStackDef([]e2e.Ec2VMOption{ec2vm.WithArch(os.WindowsOS, commonos.AMD64Arch)}))
+}
+
+func (s *vmFakeintakeSuite) TestVM() {
+	output := s.Env().VM.Execute("ls")
+	require.NotEmpty(s.T(), output)
+}
+
+func (s *vmFakeintakeSuite) TestAgent() {
+	output := s.Env().Agent.Status()
+
+	require.Contains(s.T(), output.Content, "Getting the status from the agent")
+	err := s.Env().Agent.WaitForReady()
+	require.NoError(s.T(), err)
+	isReady, err := s.Env().Agent.IsReady()
+	require.NoError(s.T(), err)
+	assert.True(s.T(), isReady, "Agent is not ready")
+}
+
+func (s *vmFakeintakeSuite) TestMetrics() {
+	t := s.T()
+	err := backoff.Retry(func() error {
```

Comment:
Out of the scope of this PR: I think it would be great to provide a function that gets metrics and based on a predicate, retry while the predicate returns false. It would make this code easier.

Reply:
Inside `fakeintake` ? Yes we can do tha

Reply:
Creates task AETT-152